bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,139 @@
+        await schedule_user.remove_roles(discord.Object(Roles.video), reason="Streaming access revoked")
+
+    async def _add_to_redis_cache(self, user_id: int, timestamp: float) -> None:
+        """Adds 'task' to redis cache."""
+        await self.task_cache.set(user_id, timestamp)
+
+    async def _reload_tasks_from_redis(self) -> None:
+        await self.bot.wait_until_guild_available()
+        items = await self.task_cache.items()
+        for key, value in items:
+            member = await self.bot.get_guild(Guild.id).fetch_member(key)
+            self.scheduler.schedule_at(datetime.datetime.utcfromtimestamp(value),
+                                       key,
+                                       self._remove_streaming_permission(member))
+
+    async def _delete_from_redis(self, key: str) -> None:
+        await self.task_cache.delete(key)
+
+    @commands.command(aliases=("streaming",))
+    @commands.has_any_role(*STAFF_ROLES)
+    async def stream(
+            self,
+            ctx: commands.Context,
+            user: discord.Member,
+            duration: Expiry = None,
+            *_
+    ) -> None:
+        """
+        Temporarily grant streaming permissions to a user for a given duration.
+
+        A unit of time should be appended to the duration.
+        Units (∗case-sensitive):
+        \u2003`y` - years
+        \u2003`m` - months∗
+        \u2003`w` - weeks
+        \u2003`d` - days
+        \u2003`h` - hours
+        \u2003`M` - minutes∗
+        \u2003`s` - seconds
+
+        Alternatively, an ISO 8601 timestamp can be provided for the duration.
+        """
+        # if duration is none then calculate default duration
+        if duration is None:
+            now = datetime.datetime.utcnow()
+            duration = now + datetime.timedelta(minutes=VideoPermission.default_permission_duration)
+
+        # Check if user already has streaming permission
+        already_allowed = any(Roles.video == role.id for role in user.roles)
+        if already_allowed:
+            await ctx.send(f"{Emojis.cross_mark} This user can already stream.")
+            return
+
+        # Schedule task to remove streaming permission from Member and add it to task cache
+        self.scheduler.schedule_at(duration, user.id, self._remove_streaming_permission(user))
+        await self._add_to_redis_cache(user.id, duration.timestamp())
+        await user.add_roles(discord.Object(Roles.video), reason="Temporary streaming access granted")
+        duration = format_infraction_with_duration(str(duration))
+        await ctx.send(f"{Emojis.check_mark} {user.mention} can now stream until {duration}.")
+
+    @commands.command(aliases=("pstream",))
+    @commands.has_any_role(*STAFF_ROLES)
+    async def permanentstream(
+            self,
+            ctx: commands.Context,
+            user: discord.Member,
+            *_
+    ) -> None:
+        """Permanently give user a streaming permission."""
```

Comment:
I find "streaming permission" to be an awkward term.
```suggestion
        """Permanently grant a user the permission to stream."""
```